apis/kueue/webhooks/workload_webhook.go

```diff
@@ -14,67 +14,75 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package webhooks
 
 import (
+	"context"
+
 	apivalidation "k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
-)
 
-const (
-	DefaultPodSetName = "main"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/constants"
 )
 
 // log is for logging in this package.
 var workloadlog = ctrl.Log.WithName("workload-webhook")
 
-func (r *Workload) SetupWebhookWithManager(mgr ctrl.Manager) error {
+type WorkloadWebhook struct{}
+
+func setupWebhookForWorkload(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
+		For(&kueue.Workload{}).
+		WithDefaulter(&WorkloadWebhook{}).
+		WithValidator(&WorkloadWebhook{}).
 		Complete()
 }
 
 // +kubebuilder:webhook:path=/mutate-kueue-x-k8s-io-v1alpha1-workload,mutating=true,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=workloads,verbs=create;update,versions=v1alpha1,name=mworkload.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &Workload{}
+var _ webhook.CustomDefaulter = &WorkloadWebhook{}
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type
-func (r *Workload) Default() {
-	workloadlog.V(5).Info("defaulter", "workload", klog.KObj(r))
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *WorkloadWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	wl := obj.(*kueue.Workload)
+	workloadlog.V(5).Info("defaulter", "workload", klog.KObj(wl))
```

Comment:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message